Use content collections instead of astro.glob [issue 2945] (#3182)
withastro/docs: src/content/docs/en/guides/imports.mdx

 

[Read more about the glob pattern syntax](https://github.com/mrmlnc/fast-glob#pattern-syntax).

> #### `Astro.glob()` vs Content Collections

Reviewer: ```suggestion
#### `Astro.glob()` vs `getCollection()`
```
I think this might be the more apt comparison!

> [Content collections](/en/guides/content-collections/) were introduced in `astro@2.0.0`. It's the best way to work with Markdown and MDX files in an Astro project. If your directory contains only Markdown and MDX files, it's best to use Content collections and the `src/content` directory.

Reviewer 2: Thanks for this @donovangomez!

I’ll let @sarah11918 decide here, but I think we don’t generally tell people what is “best” but try to describe some differences and allow people to decide for themselves. Hard to do in a short paragraph where we don’t want to repeat to much of what’s in the content collections guide itself. Maybe something like this?

```suggestion
[Content collections](/en/guides/content-collections/) provide a `getCollection()` API for loading multiple files instead of `Astro.glob()`. If you are working with Markdown and MDX files, you can store them in the `src/content/` directory to take advantage of content collection features.
```

Author: Thanks @delucis ! My bad that was totally mentioned in the writing guide. I do like the rewrite a lot more, it's more neutral.

Reviewer: Yup, @delucis' edit is much more in keeping with our docs style. I'm going to adjust it just a bit, to help make it more future-proof (e.g. Markdoc...) but it's absolutely the way to go!  :raised_hands:

> [Content collections](/en/guides/content-collections/) were introduced in `astro@2.0.0`. It's the best way to work with Markdown and MDX files in an Astro project. If your directory contains only Markdown and MDX files, it's best to use Content collections and the `src/content` directory.

Reviewer 2: ✋ Can I veto “content-based files”?

I know we sometimes say “content-based websites”, but here I think we can either say “content files” or just “content”.

```suggestion
[Content collections](/en/guides/content-collections/) provide a [`getCollection()` API](/en/reference/api-reference/#getcollection) for loading multiple files instead of `Astro.glob()`. If your content files (e.g. Markdown, MDX, Markdoc) are located in collections within the `src/content/` directory, use `getCollection()` to [query a collection](/en/guides/content-collections/#querying-collections) and return content entries.
```

> [Content collections](/en/guides/content-collections/) were introduced in `astro@2.0.0`. It's the best way to work with Markdown and MDX files in an Astro project. If your directory contains only Markdown and MDX files, it's best to use Content collections and the `src/content` directory.

Reviewer 3: I don't know if this is better or worse but here I goes
```suggestion
[Content collections](/en/guides/content-collections/) is a new content layer API to fetch local content files (e.g. Markdown, MDX, Markdoc) introduced in `astro@2.0.0`. You can use `getCollection()` instead of `Astro.glob` to [query a collection](/en/guides/content-collections/#querying-collections) and return content entries.

> [Content collections](/en/guides/content-collections/) were introduced in `astro@2.0.0`. It's the best way to work with Markdown and MDX files in an Astro project. If your directory contains only Markdown and MDX files, it's best to use Content collections and the `src/content` directory.

Reviewer: Love it when Chris pulls out his veto! 

I'm going to go with that!

> [Content collections](/en/guides/content-collections/) were introduced in `astro@2.0.0`. It's the best way to work with Markdown and MDX files in an Astro project. If your directory contains only Markdown and MDX files, it's best to use Content collections and the `src/content` directory.

Reviewer: Kevin, I went with Chris' suggestion! In particular, I think we're standardizing on not mentioning how things are new when they are just "things that exist" now. So, I would have taken that part entirely out. We typically use the `added in` tag for that, so if they really care, the info is available in a way that makes them read fewer words.  :sweat_smile:

> #### Astro.glob() vs Content Collections
> 
> [Content collections](https://docs.astro.build/en/guides/content-collections/) were introduced in `astro@2.0.0`. It's the best way to work with Markdown and MDX files in an Astro project. If your directory contains only Markdown and MDX files, it's best to use Content collections and the `src/content` directory.

Reviewer 4: Fix for the absolute link
```suggestion
#### Astro.glob() vs Content Collections

[Content collections](/en/guides/content-collections/) were introduced in `astro@2.0.0`. It's the best way to work with Markdown and MDX files in an Astro project. If your directory contains only Markdown and MDX files, it's best to use Content collections and the `src/content` directory.
```

Author: Aye good looking out @Yan-Thomas. Thanks!
